src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,37 @@
+package duke;
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
+/**
+ * Represents a task with a specific deadline.
+ * Inherits from the Task class.
+ */
+public class Deadline extends Task{
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy");
+    protected LocalDate by;
+    /**
```

Comment:
Perhaps there could be an empty line before this? Also noticed this in some other places